src/realm/sync/subscriptions.cpp

```diff
@@ -296,7 +312,31 @@ bool MutableSubscriptionSet::erase(const Query& query)
     auto ptr = find(query);
     if (!ptr)
         return false;
-    m_subs.erase(m_subs.begin() + (ptr - &m_subs.front()));
+    auto it = m_subs.begin() + (ptr - &m_subs.front());
+    erase(it);
+    return true;
+}
+
+bool MutableSubscriptionSet::erase_by_class_name(StringData object_class_name)
+{
+    // TODO: Use std::erase_if when switching to C++20.
+    auto it = std::remove_if(m_subs.begin(), m_subs.end(), [&object_class_name](const Subscription& sub) {
+        return sub.object_class_name == object_class_name;
+    });
```

Comment:
This is an interesting function - I don't think I've ever used it before, but it is def needed since there could be multiple subscriptions with the same class name. It seems odd that it moves the "removed" items past the `end()` so they can be erased later.

Reply:
Yeah, it uses the so called erase-remove idiom: https://en.wikipedia.org/wiki/Erase%E2%80%93remove_idiom